server/handlers/handlers.go

```diff
@@ -32,7 +32,7 @@ func RegisterUser(w http.ResponseWriter, res *http.Request) {
 	}
 
 	err := data.CreateUser(newUser)
-	if err == nil {
+	if err != nil {
```

Comment:
Nice job catching this bug! 
Now that I think about it, should these functions should be in `handlers/user.go`? You can just rename this file tbh.

Reply:
Sounds good! I think this way we will finally modularize the different components in the backend!